### DISCUSSION TITLE
当 node_modules 根目录和子目录共同存在 antd 的依赖时，本地服务依赖路径有问题

### DISCUSSION THREAD

**Discussion by 1eeing** [umijs/umi#8746] [Q&A]
问题描述：
我的 umi 项目，node_modules 根目录下有一个 antd 4.0.0 版本，node_modules 下的子包 A 因为 dependencies 了 antd@4.18.0，因此在 node_modules/A/node_modules/ 存在一份 antd@4.18.0 。

我的疑问：
存在两份 antd 我觉得没问题，这是 npm 的一个规则。但是使用 umi 启动后，子包却是在根目录引用 antd@4.0.0，从而导致报错。

期望：
这个场景，应该从子包的 node_modules 中引用 antd@4.18.0 才对。（vscode 中提示的引用路径就是对的，但是实际启动后路径就是错的）

麻烦帮忙看看

**Comment by fz6m** [comment]
由于缺乏足够的信息，请提供 [最小重现](https://stackoverflow.com/help/minimal-reproducible-example) 的 demo 帮助排查问题。

**Reply by 1eeing** [reply]
解决了，是因为 umi 内置的 webpack resolve alias 中 antd 指向了 node_modules 根目录下的 antd 导致的，通过在 .umirc 中添加如下代码修复

```ts
// .umirc
chainWebpack(memo, args) {
    const alias = memo.resolve.alias;
    alias.delete('antd');
},
```
